Biter Corpse Removal

adrianvanderwal opened the pull request:
Adds biter corpse removal once threshold is reached

closes #608

Commit:
Biter Corpse Removal

Adds biter corpse removal once threshold is reached

mheguy reviewed (changes requested):
Please test your code (it will error on the death of any entity).
Please look at Travis CI's feedback (the feedback will indicate the cause of the first error you will run into) by checking the details:
![image](https://user-images.githubusercontent.com/10162554/54785351-67649e80-4bfc-11e9-99e2-973433f45b0a.png)

grilledham commented:
For the most part the code looks good. The typos that travis pointed out can be easily fixed. Also you should check the entity is valid. Other handlers may run on the same event and make the entity invalid, unlikely in this case but possible so you have to protect against it. 

I do however have concerns about the potential performance impact of such code. Before this is merged it would be good if someone could run some tests to try and measure the performance impact of this code. Which I'll admit wont be easy to do.

 If performance does turn out to be an issue one idea would to be early return every 99 out of 100 times the function is called. Or we could use one of your suggestions in the issue and keep a count of corpses per chunk and clear the chunk when it reaches a threshold.

Another thought is `surface.find_entities_filtered` probably returns entities from the top left to the bottom right of the search area. That might look odd when clearing, we could instead destroy every other corpse to leave some behind.

Commit:
Changes as requested

Fixed using undeclared variable before definition
Added validity checks
Removed "limit", removes half of the corpses found instead

mheguy commented on features/biter_corpse_util.lua:
```diff
@@ -15,52 +15,53 @@ Global.register(
+                    name = biter_corpses,
+                }
+            ) do
+                if corpse.valid and k % 2 == 1 then
```
`corpse` is coming from `find_entities_filtered` which will always return valid results, so no need to check validity here

mheguy reviewed (changes requested):
L19: `if entity.valie then` suggests the code is still untested. Please inform us when you have tested this PR.

Commit:
Requested Changes and Testing

adrianvanderwal commented:
I have tested this PR on my local machine - barely slows the game down when used on crash site.
The removal of half the corpses is very sudden - may need some tweaking to find numbers that are not so drastic.
